Chat

molro opened the pull request:
Corrección

Instrucciones Readme.md
Actualizadas

molro committed:
Update Project

molro committed:
Estructura del proyecto

molro committed:
Enlaces

molro committed:
Upd

molro committed:
Rutas creadas & Requirimientos en Readme

molro committed:
Updated Gitingore

molro committed:
Endpoint Users finalizado (N1Ex1)

molro committed:
Nivell 1 - Exercise 2

molro committed:
Update README.md

molro committed:
Update readme

molro committed:
Updrdm

molro committed:
Update README.md

molro committed:
Update Package.json

molro committed:
Merge branch 'sprint41' of https://github.com/molivaresr/nodeInitialDemo into sprint41

molro committed:
uploads

molro committed:
uploads

molro committed:
Niveles 1 al 3

Colección de Postman incluída

molro committed:
Chat - Server (running)

molro committed:
Diseño de Login

molro committed:
Front v2

Server Alternativo

molro committed:
Front V2.3 / Back Socket conectado

molro committed:
Upd Repo

molro committed:
Login test

molro committed:
Merge branch 'chat' of https://github.com/molro/nodeInitialDemo into chat

molro committed:
Servidor + Orden de Front

molro committed:
Reorganización

Server  (Conectado Sockets)
Front  Reorganizado

molro committed:
TS Client Working

molro committed:
Reorganización

molro committed:
Eliminando

molro committed:
useRef solucionap

molro committed:
Git Ignore updated

molro committed:
gitignore

molro committed:
Inicio de sesion

molro committed:
icloud

molro committed:
Creando salas

Pendiente renderizado en el front end

molro committed:
Pendiente de inicializar en NEXT

molro committed:
Renderizando Rooms

Pendiente

molro committed:
Recibiendo salas del server!

molro committed:
Merge branch 'chat' of https://github.com/molro/nodeInitialDemo into chat

molro committed:
Leyendo Rooms desde el server

molro committed:
Casi Casi

molro committed:
Renderizado OK! - PTE REFACTORIZAR FUNCION

molro committed:
Refactor y Maquetación

molro committed:
Server

molro committed:
Front

molro committed:
Utilizando Context

:D

molro committed:
Mensajes

molro committed:
Create login.css

Login Hacked

molro committed:
NewSockets

molro committed:
Merge branch 'chat' of https://github.com/molro/nodeInitialDemo into chat

molro committed:
Front Working

molro committed:
Mensajes y error detectado.

molro committed:
Persistencia Mongo

molro committed:
Server / Rutas / Persistencia

molro committed:
TokensAdvance

molro committed:
Limpiando Importaciones

molro committed:
Testeando JWT y limpiando rutas

molro committed:
Token. Pendiente validar

molro committed:
Validaciones y persistencia

molro committed:
Login OK

Token recibido!
Comenzando a cargar información desde BD

molro committed:
Persistencia Mensajes y Salas

:D

molro committed:
Features 80%

Sockets Back y Front Conectados
Persistencia Ok
10% Front  details
10% Estilos CSS

molro committed:
Detalles

Server - Update Model Users
Front - Listando usuarios en cada sala

molro committed:
Login

Recovery Login & Register

molro committed:
Login / rutas

molro committed:
Casi Login

molro committed:
Login

Login Working

molro committed:
Login Ok! Rutas, Registro

molro committed:
Registro Pulido

molro committed:
Debugging sockets

molro committed:
Perfilando

molro committed:
Solución de Errores

Correción de bug!
Problema con importación de Layout y llamado a los componentes

molro committed:
Upd

molro committed:
Pendiente Obtener Listado de usuarios

molro committed:
Listados!

molro committed:
Cleaning Repo

molro committed:
Recibiendo Usuarios

molro committed:
Estilos

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
entrega final

molro committed:
Merge branch 'chat' of https://github.com/molro/nodeInitialDemo into chat

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Update readme.md

molro committed:
Entrega final

molro committed:
Merge branch 'chat' of https://github.com/molro/nodeInitialDemo into chat

molro committed:
title

molro committed:
titulo

StratocasterO commented:
Mauricio, he tenido dos errores y no puedo probarlo:

- El primero al arrancar la app de react. Me ha salido esto, pero he buscado y forzando el guardado del `package.json` se ha arreglado solo:
![image](https://user-images.githubusercontent.com/32362743/166118855-d9e64a0d-20e6-488f-b3c5-48d25dacf3fb.png)

- El segundo no me deja registrar usuarios. La consola del explorador dice esto:
![image](https://user-images.githubusercontent.com/32362743/166118873-9b7721d9-08ec-4586-9094-6c95f157fd5b.png)

Dale un ojo y el lunes lo volvemos a mirar 😉

molro committed:
Corregido Error

molro committed:
Cambios Menores

molro committed:
Login fixed

StratocasterO commented:
Problemas que arreglar:

- La página se recarga al hacer login pero no me lleva al chat
- Aunque hago logout con un usuario sigue apareciendo en la lista de usuarios conectados
- La sala principal no tiene permanencia de mensajes (o añades permanencia o no permites escribir, una de dos)
- Me salen los mismos usuarios conectados en todas las salas

A parte de estos problemas funcionales, algunas buenas prácticas del back:

- Logs excesivos en el server, no hace falta mostrar por la consola cada intento de conexión, mensaje y cambio de sala
- Tienes trozos de código comentados por todas partes, limpia un poco esas cosas o haz una rama "limpia" y otra de trabajo

Dale un repaso, tienes que acabar esta semana sin falta! 🔥

molro committed:
Mejorando Usuarios

molro committed:
Cambios Finales

molro committed:
Corrección de errores

Login - Redirige a Chat
Logout - Quita usuarios
Listado de usuarios - Coordinado con la sala
Sala principal - No permite escribir

- Limpieza de código

StratocasterO commented:
Buen trabajo, Mauricio, está mucho mejor! Algunas cosas a tener en cuenta:

- Bien hecho con la estructura del server y buen trabajo con TypeScript ✔
- Sobre el back, sigues teniendo muchos logs que deberían estar solo en `dev`, no en `prod`
- También habría que limpiar un poco algunas cosas: imports que no se usan, diversificar un poco más algunos archivos que son demasiado extensos, etc 🧹
- También veo muchos endpoints, parece que te has complicado más de la cuenta con eso (habría que hacer el máximo posible con sockets)

- Buen trabajo también haciendo el front en React, aun con la pinta ochentera 😜
- Es muy importante añadir verificaciones de email, contraseña y demás, ahora mismo acepta cualquier string como email
- También hay que comunicar al usuario de manera clara cualquier error: contraseña incorrecta, email no válido, login fallido...
- Ojo también con los logs en el front, también hay que limitarlos en `prod`

Acaba entre mañana y el finde las dos entregas de estructura de datos y el ejercicio de los dados para tenerlo corregido el lunes y hacer la defensa. Vamos, que ya lo tienes! 🔥

molro committed:
Update readme.md

molro committed:
Update readme.md
